diff --git a/build/libv86-debug.js b/build/libv86-debug.js
index f491f64c1390eb8eec5e6a7265381df43b3cbc3e..0fa51e183ecfc748f3cecf176365b2f311c90afe 100644
--- a/build/libv86-debug.js
+++ b/build/libv86-debug.js
@@ -5384,7 +5384,8 @@ $FS$$module$lib$filesystem$$.prototype.CreateSymlink = function($filename$jscomp
   $parent_inode$jscomp$5_x$jscomp$114$$.gid = this.inodes[$foreign_parentid$jscomp$3_parentid$jscomp$9$$].gid;
   $parent_inode$jscomp$5_x$jscomp$114$$.qid.type = 160;
   $parent_inode$jscomp$5_x$jscomp$114$$.symlink = $symlink$$;
-  $parent_inode$jscomp$5_x$jscomp$114$$.mode = 40960;
+  $parent_inode$jscomp$5_x$jscomp$114$$.mode = 41471;
+  $parent_inode$jscomp$5_x$jscomp$114$$.size = $symlink$$.length;
   this.PushInode($parent_inode$jscomp$5_x$jscomp$114$$, $foreign_parentid$jscomp$3_parentid$jscomp$9$$, $filename$jscomp$9_foreign_id$jscomp$3$$);
   return this.inodes.length - 1;
 };
diff --git a/build/libv86-debug.mjs b/build/libv86-debug.mjs
index 0d675b5c7bcfbd0d12132d73de5e5b9d9b34f097..358b4d8c7d9a85c16cb9147c1016a68662d49cfc 100644
--- a/build/libv86-debug.mjs
+++ b/build/libv86-debug.mjs
@@ -5383,7 +5383,8 @@ $FS$$module$lib$filesystem$$.prototype.CreateSymlink = function($filename$jscomp
   $parent_inode$jscomp$5_x$jscomp$114$$.gid = this.inodes[$foreign_parentid$jscomp$3_parentid$jscomp$9$$].gid;
   $parent_inode$jscomp$5_x$jscomp$114$$.qid.type = 160;
   $parent_inode$jscomp$5_x$jscomp$114$$.symlink = $symlink$$;
-  $parent_inode$jscomp$5_x$jscomp$114$$.mode = 40960;
+  $parent_inode$jscomp$5_x$jscomp$114$$.mode = 41471;
+  $parent_inode$jscomp$5_x$jscomp$114$$.size = $symlink$$.length;
   this.PushInode($parent_inode$jscomp$5_x$jscomp$114$$, $foreign_parentid$jscomp$3_parentid$jscomp$9$$, $filename$jscomp$9_foreign_id$jscomp$3$$);
   return this.inodes.length - 1;
 };
diff --git a/build/libv86.js b/build/libv86.js
index bcf1b646273db63c5b76daf1eff2522b87633f24..8809aebc593755e2aa0165253a2a04dc1558704e 100644
--- a/build/libv86.js
+++ b/build/libv86.js
@@ -302,7 +302,7 @@ this.inodedata[c];delete this.inodedata[c];d.direntries=new Map;d.nlinks=0;retur
 O.prototype.CreateDirectory=function(a,b){var c=this.inodes[b];if(0<=b&&this.is_forwarder(c))return b=c.foreign_id,a=this.follow_fs(c).CreateDirectory(a,b),this.create_forwarder(c.mount_id,a);c=this.CreateInode();c.mode=16895;0<=b&&(c.uid=this.inodes[b].uid,c.gid=this.inodes[b].gid,c.mode=this.inodes[b].mode&511|16384);c.qid.type=64;this.PushInode(c,b,a);this.NotifyListeners(this.inodes.length-1,"newdir");return this.inodes.length-1};
 O.prototype.CreateFile=function(a,b){var c=this.inodes[b];if(this.is_forwarder(c))return b=c.foreign_id,a=this.follow_fs(c).CreateFile(a,b),this.create_forwarder(c.mount_id,a);c=this.CreateInode();c.uid=this.inodes[b].uid;c.gid=this.inodes[b].gid;c.qid.type=128;c.mode=this.inodes[b].mode&438|32768;this.PushInode(c,b,a);this.NotifyListeners(this.inodes.length-1,"newfile");return this.inodes.length-1};
 O.prototype.CreateNode=function(a,b,c,d){var e=this.inodes[b];if(this.is_forwarder(e))return b=e.foreign_id,a=this.follow_fs(e).CreateNode(a,b,c,d),this.create_forwarder(e.mount_id,a);e=this.CreateInode();e.major=c;e.minor=d;e.uid=this.inodes[b].uid;e.gid=this.inodes[b].gid;e.qid.type=192;e.mode=this.inodes[b].mode&438;this.PushInode(e,b,a);return this.inodes.length-1};
-O.prototype.CreateSymlink=function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,a=this.follow_fs(d).CreateSymlink(a,b,c),this.create_forwarder(d.mount_id,a);d=this.CreateInode();d.uid=this.inodes[b].uid;d.gid=this.inodes[b].gid;d.qid.type=160;d.symlink=c;d.mode=40960;this.PushInode(d,b,a);return this.inodes.length-1};
+O.prototype.CreateSymlink=function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,a=this.follow_fs(d).CreateSymlink(a,b,c),this.create_forwarder(d.mount_id,a);d=this.CreateInode();d.uid=this.inodes[b].uid;d.gid=this.inodes[b].gid;d.qid.type=160;d.symlink=c;d.mode=41471;d.size=c.length;this.PushInode(d,b,a);return this.inodes.length-1};
 O.prototype.CreateTextFile=async function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,c=await this.follow_fs(d).CreateTextFile(a,b,c),this.create_forwarder(d.mount_id,c);d=this.CreateFile(a,b);b=this.inodes[d];a=new Uint8Array(c.length);b.size=c.length;for(b=0;b<c.length;b++)a[b]=c.charCodeAt(b);await this.set_data(d,a);return d};
 O.prototype.CreateBinaryFile=async function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,c=await this.follow_fs(d).CreateBinaryFile(a,b,c),this.create_forwarder(d.mount_id,c);d=this.CreateFile(a,b);a=this.inodes[d];b=new Uint8Array(c.length);b.set(c);await this.set_data(d,b);a.size=c.length;return d};O.prototype.OpenInode=async function(a,b){var c=this.inodes[a];if(this.is_forwarder(c))return await this.follow_fs(c).OpenInode(c.foreign_id,b);16384===(c.mode&61440)&&this.FillDirectory(a)};
 O.prototype.CloseInode=async function(a){var b=this.inodes[a];if(this.is_forwarder(b))return await this.follow_fs(b).CloseInode(b.foreign_id);2===b.status&&this.storage.uncache(b.sha256sum);4===b.status&&(b.status=-1,await this.DeleteData(a))};
diff --git a/build/libv86.mjs b/build/libv86.mjs
index 8d22df0e352578d3bbe6bb9863c59db756b1a74d..84a82eb9064496cc820567555ef66f4829e6cb1a 100644
--- a/build/libv86.mjs
+++ b/build/libv86.mjs
@@ -268,7 +268,7 @@ this.qid.version=a[11];this.qid.path=a[12];this.nlinks=a[13]};O.prototype.divert
 this.inodes[h].direntries.set("..",f);this.inodedata[f]=this.inodedata[c];delete this.inodedata[c];d.direntries=new Map;d.nlinks=0;return f};O.prototype.copy_inode=function(a,b){Object.assign(b,a,{fid:b.fid,direntries:b.direntries,nlinks:b.nlinks})};O.prototype.CreateInode=function(){const a=Math.round(Date.now()/1E3),b=new Cc(++this.qidcounter.last_qidnumber);b.atime=b.ctime=b.mtime=a;return b};O.prototype.CreateDirectory=function(a,b){var c=this.inodes[b];if(0<=b&&this.is_forwarder(c))return b=
 c.foreign_id,a=this.follow_fs(c).CreateDirectory(a,b),this.create_forwarder(c.mount_id,a);c=this.CreateInode();c.mode=16895;0<=b&&(c.uid=this.inodes[b].uid,c.gid=this.inodes[b].gid,c.mode=this.inodes[b].mode&511|16384);c.qid.type=64;this.PushInode(c,b,a);this.NotifyListeners(this.inodes.length-1,"newdir");return this.inodes.length-1};O.prototype.CreateFile=function(a,b){var c=this.inodes[b];if(this.is_forwarder(c))return b=c.foreign_id,a=this.follow_fs(c).CreateFile(a,b),this.create_forwarder(c.mount_id,
 a);c=this.CreateInode();c.uid=this.inodes[b].uid;c.gid=this.inodes[b].gid;c.qid.type=128;c.mode=this.inodes[b].mode&438|32768;this.PushInode(c,b,a);this.NotifyListeners(this.inodes.length-1,"newfile");return this.inodes.length-1};O.prototype.CreateNode=function(a,b,c,d){var e=this.inodes[b];if(this.is_forwarder(e))return b=e.foreign_id,a=this.follow_fs(e).CreateNode(a,b,c,d),this.create_forwarder(e.mount_id,a);e=this.CreateInode();e.major=c;e.minor=d;e.uid=this.inodes[b].uid;e.gid=this.inodes[b].gid;
-e.qid.type=192;e.mode=this.inodes[b].mode&438;this.PushInode(e,b,a);return this.inodes.length-1};O.prototype.CreateSymlink=function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,a=this.follow_fs(d).CreateSymlink(a,b,c),this.create_forwarder(d.mount_id,a);d=this.CreateInode();d.uid=this.inodes[b].uid;d.gid=this.inodes[b].gid;d.qid.type=160;d.symlink=c;d.mode=40960;this.PushInode(d,b,a);return this.inodes.length-1};O.prototype.CreateTextFile=async function(a,b,c){var d=this.inodes[b];
+e.qid.type=192;e.mode=this.inodes[b].mode&438;this.PushInode(e,b,a);return this.inodes.length-1};O.prototype.CreateSymlink=function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,a=this.follow_fs(d).CreateSymlink(a,b,c),this.create_forwarder(d.mount_id,a);d=this.CreateInode();d.uid=this.inodes[b].uid;d.gid=this.inodes[b].gid;d.qid.type=160;d.symlink=c;d.mode=41471;d.size=c.length;this.PushInode(d,b,a);return this.inodes.length-1};O.prototype.CreateTextFile=async function(a,b,c){var d=this.inodes[b];
 if(this.is_forwarder(d))return b=d.foreign_id,c=await this.follow_fs(d).CreateTextFile(a,b,c),this.create_forwarder(d.mount_id,c);d=this.CreateFile(a,b);b=this.inodes[d];a=new Uint8Array(c.length);b.size=c.length;for(b=0;b<c.length;b++)a[b]=c.charCodeAt(b);await this.set_data(d,a);return d};O.prototype.CreateBinaryFile=async function(a,b,c){var d=this.inodes[b];if(this.is_forwarder(d))return b=d.foreign_id,c=await this.follow_fs(d).CreateBinaryFile(a,b,c),this.create_forwarder(d.mount_id,c);d=this.CreateFile(a,
 b);a=this.inodes[d];b=new Uint8Array(c.length);b.set(c);await this.set_data(d,b);a.size=c.length;return d};O.prototype.OpenInode=async function(a,b){var c=this.inodes[a];if(this.is_forwarder(c))return await this.follow_fs(c).OpenInode(c.foreign_id,b);16384===(c.mode&61440)&&this.FillDirectory(a)};O.prototype.CloseInode=async function(a){var b=this.inodes[a];if(this.is_forwarder(b))return await this.follow_fs(b).CloseInode(b.foreign_id);2===b.status&&this.storage.uncache(b.sha256sum);4===b.status&&
 (b.status=-1,await this.DeleteData(a))};O.prototype.Rename=async function(a,b,c,d){if(a===c&&b===d)return 0;var e=this.Search(a,b);if(-1===e)return-2;var f=this.GetFullPath(a)+"/"+b;if(-1!==this.Search(c,d)){var g=this.Unlink(c,d);if(0>g)return g}var h=this.inodes[e],m=this.inodes[a];g=this.inodes[c];if(this.is_forwarder(m)||this.is_forwarder(g))if(this.is_forwarder(m)&&m.mount_id===g.mount_id){if(a=await this.follow_fs(m).Rename(m.foreign_id,b,g.foreign_id,d),0>a)return a}else{if(this.is_a_root(e)||
